x-pack/legacy/plugins/monitoring/public/components/alerts/manage_email_action.tsx

```diff
@@ -0,0 +1,301 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { Fragment } from 'react';
+import {
+  EuiForm,
+  EuiFormRow,
+  EuiFieldText,
+  EuiLink,
+  EuiSpacer,
+  EuiFieldNumber,
+  EuiFieldPassword,
+  EuiSwitch,
+  EuiButton,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiSuperSelect,
+  EuiText,
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { ActionResult } from '../../../../actions/server/types';
+import { getMissingFieldErrors, hasErrors, getRequiredFieldError } from '../../lib/form_validation';
+import { ALERT_EMAIL_SERVICES } from '../../../common/constants';
+
+export interface EmailActionData {
+  service: string;
+  host: string;
+  port: number | ''; // support a string to ensure the user can backspace to an empty field
```

Comment:
Seems kinda strange that this value can be either of type number or empty str. Maybe just making it optional is enough?
```javascript
port?: number
//or maybe...
port: number | void
```

I might be wrong here though

Reply:
Yup, I can do that!